wp-redis.php

```diff
@@ -36,19 +38,19 @@ function wp_redis_get_info() {
 
 	if ( empty( $redis_server ) ) {
 		// Attempt to automatically load Pantheon's Redis config from the env.
-		if ( isset( $_SERVER['CACHE_HOST'] ) ) {
-			$redis_server = array(
-				'host'     => $_SERVER['CACHE_HOST'],
-				'port'     => $_SERVER['CACHE_PORT'],
-				'auth'     => $_SERVER['CACHE_PASSWORD'],
-				'database' => isset( $_SERVER['CACHE_DB'] ) ? $_SERVER['CACHE_DB'] : 0,
-			);
+		if ( isset( $_SERVER['CACHE_HOST'] ) && isset( $_SERVER['CACHE_PORT'] ) && isset( $_SERVER['CACHE_PASSWORD'] ) && isset( $_SERVER['CACHE_DB'] ) ) {
+			$redis_server = [
+				'host' => sanitize_text_field( $_SERVER['CACHE_HOST'] ),
+				'port' => sanitize_text_field( $_SERVER['CACHE_PORT'] ),
+				'auth' => sanitize_text_field( $_SERVER['CACHE_PASSWORD'] ),
+				'database' => sanitize_text_field( $_SERVER['CACHE_DB'] ),
```

Comment:
@jazzsequence this was also incorrectly changed to no longer be providing a default value.